src/MiddlewareListener.php

```diff
@@ -96,6 +117,42 @@ public function onDispatch(MvcEvent $event)
     }
 
     /**
+     * Create a middleware pipe from the array spec given.
+     *
+     * @param ContainerInterface $serviceLocator
+     * @param ResponseInterface $responsePrototype
+     * @param array $middlewaresToBePiped
+     * @return MiddlewarePipe
+     * @throws \InvalidArgumentException
+     * @throws \Zend\Mvc\Exception\MiddlewareNotCallableException
+     */
+    private function createPipeFromSpec(
+        ContainerInterface $serviceLocator,
+        ResponseInterface $responsePrototype,
+        array $middlewaresToBePiped
+    ) {
+        $pipe = new MiddlewarePipe();
+        $pipe->setResponsePrototype($responsePrototype);
+        foreach ($middlewaresToBePiped as $middlewareToBePiped) {
+            if (null === $middlewareToBePiped) {
+                throw new \InvalidArgumentException('Middleware name cannot be null');
+            }
+
+            $middlewareName = is_string($middlewareToBePiped) ? $middlewareToBePiped : get_class($middlewareToBePiped);
+
+            if (is_string($middlewareToBePiped) && $serviceLocator->has($middlewareToBePiped)) {
+                $middlewareToBePiped = $serviceLocator->get($middlewareToBePiped);
+            }
+            if (! is_callable($middlewareToBePiped)) {
```

Comment:
This needs to change to be:

```php
if (! $middlewareToBePiped instanceof \Http\Interop\ServerMiddleware\MiddlewareInterface
    && ! is_callable($middlewareToBePiped)
)
```

as zend-stratigility 2.0 allows either of those types. Testing only for `is_callable()` means that http-interop middleware cannot be used!